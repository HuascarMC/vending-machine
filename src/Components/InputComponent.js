import React from 'react';
import BalanceDisplayComponent from './BalanceDisplayComponent'

class InputComponent extends React.Component {
  render() {
    return(
      <div>
        <BalanceDisplayComponent balance={ this.props.balance } />
        <div className="input-wrapper">
        <ul className="input-box">
          <li><p className="coin-value">$0.01<button onClick={ ()=>this.props.updateBalance(0.01) } className="coin bronze"></button></p></li>
          <li><p className="coin-value">$0.05<button onClick={ ()=>this.props.updateBalance(0.05) } className="coin silver"></button></p></li>
          <li><p className="coin-value">$0.10<button onClick={ ()=>this.props.updateBalance(0.1) } className="coin silver"></button></p></li>
          <li><p className="coin-value">$0.25<button onClick={ ()=>this.props.updateBalance(0.25) } className="coin silver"></button></p></li>
          <li><p className="coin-value">$1.00<button onClick={ ()=>this.props.updateBalance(1.00) } className="coin gold"></button></p></li>
        </ul>
      </div>
      </div>
    );
  }
}

export default InputComponent;
