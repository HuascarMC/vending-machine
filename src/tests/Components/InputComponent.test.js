import React from 'react';
import InputComponent from '../../Components/InputComponent'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
  shallow(<InputComponent />);
});

it('should render input-wrapper div', () => {
  const wrapper = shallow(<InputComponent />);
  const div = wrapper.find('.input-wrapper');
  expect(div.exists()).toEqual(true);
});

it('should render input-box list', ()=> {
  const wrapper = shallow(<InputComponent />);
  const div = wrapper.find('.input-box');
  expect(div.exists()).toEqual(true);
});

it('should render five list tags', () => {
  const wrapper = shallow(<InputComponent />);
  const li = wrapper.find('li');
  expect(li.length).toBe(5);
});

it('should render p tags in list tags', () => {
  const wrapper = shallow(<InputComponent />);
  const p = wrapper.find('li').find('p');
  expect(p.length).toBe(5);
});

it('should render p tag with penny value', () => {
  const wrapper = shallow(<InputComponent />);
  const p = wrapper.find('p');
  expect(p.at(0).text()).toEqual('$0.01');
});

it('should render p tag with nickel value', () => {
  const wrapper = shallow(<InputComponent />);
  const p = wrapper.find('p');
  expect(p.at(1).text()).toEqual('$0.05');
});

it('should render p tag with dime value', () => {
  const wrapper = shallow(<InputComponent />);
  const p = wrapper.find('p');
  expect(p.at(2).text()).toEqual('$0.10');
});

xit('should match the snapshot', () => {
  const component = renderer.create(<InputComponent />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
