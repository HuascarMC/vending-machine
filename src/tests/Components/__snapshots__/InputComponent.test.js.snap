// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    className="input-wrapper"
  >
    <ul
      className="input-box"
    >
      <li>
        <p
          className="coin-value"
        >
          $0.01
          <button
            className="coin bronze"
            onClick={[Function]}
          />
        </p>
      </li>
      <li>
        <p
          className="coin-value"
        >
          $0.05
          <button
            className="coin silver"
            onClick={[Function]}
          />
        </p>
      </li>
      <li>
        <p
          className="coin-value"
        >
          $0.10
          <button
            className="coin silver"
            onClick={[Function]}
          />
        </p>
      </li>
      <li>
        <p
          className="coin-value"
        >
          $0.25
          <button
            className="coin silver"
            onClick={[Function]}
          />
        </p>
      </li>
      <li>
        <p
          className="coin-value"
        >
          $1.00
          <button
            className="coin gold"
            onClick={[Function]}
          />
        </p>
      </li>
    </ul>
  </div>
</div>
`;
